py/miniconf-mqtt/miniconf/miniconf.py

```diff
@@ -95,6 +147,48 @@ async def command(self, path, value, retain=False, timeout=5):
             response_topic=self.response_topic,
             correlation_data=request_id)
 
-        result = await asyncio.wait_for(fut, timeout)
+        result = await fut
         if result['code'] != 0:
             raise MiniconfException(result['msg'])
+
+
+    def list_paths(self):
+        """ Get a list of all the paths available on the device. """
+        fut = asyncio.get_running_loop().create_future()
+
+        request_id = uuid.uuid1().hex.encode()
+        assert request_id not in self.inflight
+        self.inflight[request_id] = ([], fut)
+
+        self.client.publish(f'{self.prefix}/list', payload='',
+                            correlation_data=request_id,
+                            response_topic=self.response_topic)
+        return fut
+
+
+    async def get(self, path):
+        """ Get the specific value of a given path. """
+        fut = asyncio.get_running_loop().create_future()
+
+        # Assign unique correlation data for response dispatch
+        request_id = uuid.uuid1().hex.encode()
+        assert request_id not in self.inflight
+        self.inflight[request_id] = [None, fut]
+
+        subscription = self.client.subscribe(f'{self.prefix}/settings/{path}', no_local=True)
+        self._pending_subscriptions[subscription] = asyncio.get_running_loop().create_future()
+        await self._pending_subscriptions[subscription]
+
+        self.client.publish(
+            f'{self.prefix}/settings/{path}', payload='', qos=0,
+            response_topic=self.response_topic,
+            correlation_data=request_id)
+
+        response, value = await fut
+
+        self.client.unsubscribe(f'{self.prefix}/settings/{path}')
```

Comment:
There's an ack for it, but I figured it wasn't all that important. In reality, the ACK will come relatively quick, and the only error case would be calling `get()` multiple times with the same path, in which case the first `subscribe` will await, and we'll process the unsuback.

I don't think there's really any harm in exiting before it's ACKed

Reply:
But with asyncio I can imagine the following message sequence being sent (two concurrent get on the same topic): "sub1,pub1,time passes, sub2,suback2,resp1,unsub1,unsuback1,time passes pub2", i.e. an overlap due to asyncio and "pub2" not being seen as "unsub1" went in after "sub2".

Reply:
In fact this may be a big problem and waiting for unsuback.would.not help.
What about subscribing to all settings on creation and then never unsubscribing?

Reply:
In your scenario, it's important to note that `sub1` and `sub2` are two unique subscriptions. unsub1 does not result in cancelling out `sub2` even if they're on the same topics because they use unique identifiers to denote which subscription is being canceled.

Reply:
How does `self.client.unsubscribe(f'{self.prefix}/settings/{path}')` know which one to unsubscribe?

Reply:
Oh huh, that's a really good point. It doesn't look like GMQTT distinguishes. I'll get this fixed up

Reply:
I would still do a wildcard subscribe to all settings and remain subscribed. Seems easier to me an fine w.r.t. traffic.

Reply:
I've updated to just subscribe to the settings/# path and ignore local traffic. It will get undesired traffic from e.g. other miniconf host-side clients, but it's not problematic (log and ignore). The unsubscribe + unsub ack actually added a lot of latency during the settings tree retrieval because so many more messages were being transferred.